setup.py

```diff
@@ -15,7 +15,11 @@
     history = history_file.read().replace('.. :changelog:', '')
 
 requirements = [
-    'docopt', 'yarg', 'nbconvert', 'ipython'
+    'docopt', 'yarg'
+]
+
+jupyter_requirements = [
```

Comment:
please eli5 how this works or link a reference haha

Reply:
Here's a reference on how it works: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

Reply:
please add instructions to read me about how to deal with these optional dependencies

Reply:
also, if possible please reverse the behaviour so that by default the jupyter stuff is added and it is only excluded as a dependency when explicitly told to

Reply:
> also, if possible please reverse the behaviour so that by default the jupyter stuff is added and it is only excluded as a dependency when explicitly told to

It seems like there's no support for doing such thing. Check https://stackoverflow.com/questions/76376575/python-setuptools-exclude-dependencies-when-installing

What do you suggest doing in this case? Drop optional support? Maybe keep it as it is?

Reply:
> What do you suggest doing in this case? Drop optional support? Maybe keep it as it is?

@alan-barzilay

Reply:
hi, sorry for the delay. If there is no support for this feature, there is no support for it, what can we do haha
But i liked that `pip install --no-deps my_package` workaround, if we only import nbconverter/ipython when we use them (aka when we don't pass the cli option to ignore notebooks), someone that does this should be able to tailor their one dependencies without too much trouble. Maybe we should add a mention to this into the readme as well

Reply:
and, maybe I missed it but where do we use ipython directly? I found the import for nbvconvert, is it that ipython is a dependency of nbconvert? if thats the case, cant we omit it from the file and let nbconvert install its dependencies itself?

Reply:
please test if we really need the ipython dependency for  magics translation